### DISCUSSION TITLE
How to dynamic iOS SDK in Gymfile

### DISCUSSION THREAD

**Discussion by heart** [fastlane/fastlane#17670] [Q&A] [answered]
Hello All,

I use `fastlane gym` for my macbook and circle ci 
And every time i want to build on my macbook I need to change ios sdk in Gymfile to new lastest version
Then I need to change it back every time to the sdk version that circle ci supported when i want to push my code

how to dynamic it?

**Comment by rogerluan** [comment, marked as answer]
Hey @heart 👋 

That may be happening because the Xcode version you're using is different between your local machine and the CI - is it? You could try standardizing the Xcode versions to mitigate that problem.
Hope this helps.

Nice username btw 😎

**Reply by heart** [reply]
Thankyou @rogerluan 

I using https://circleci.com/ as my CI/CD for mobile app

The problem is circleci not support the latest version of xcode yet
But on my local I need to update xcode to the latest version to test on the latest iOS version

By the way i created my own shell script to update Gymfile
```
foo=$(xcodebuild -showsdks | grep "iphoneos" 2>&1 | head -n 1)
BUFFER=""
for (( i=0; i<${#foo}; i++ )); do
  CHAR="${foo:$i:1}"
  if [ "$CHAR" = " " ]; then
    BUFFER=""
  else 
    BUFFER="$BUFFER$CHAR"
  fi
done

SDK="$BUFFER"

sed -i '.bck' -e "s/iphoneos[0-9]*[.][0-9]*/$SDK/" ./fastlane/Gymfile
```

**Reply by rogerluan** [reply]
JW, are you on Xcode newer than 12.2? Otherwise, support to Xcode 12.2 was added on Nov 6th: https://circleci.com/docs/2.0/testing-ios/

**Reply by heart** [reply]
@rogerluan I don't like to do the same thing everytime xcode have a new version It's waste my time
I want to config once and then my team can use it as long as they can. I don't want to facing the same problem.

**Reply by rogerluan** [reply]
Your CI and your local machine should run the same Xcode versions and SDKs, that's a general good practice.
If you **need** to run the latest version to test on your local machine, and Circle CI doesn't support that version yet, it seems to be a circle CI problem.

To answer your specific question on "how to make this change automatic (dynamic)", you could script a way to determine which environment it is on, or which SDK your environment supports, so that it automatically chooses the best SDK for you.

**Reply by heart** [reply]
@rogerluan I knew it I need to use the same xcode version
But I'm not sure you understand me.

I have no problem on the local machine because we alway update XCode to the latest version.

But for the CI/CD script 
I can't allow junior developer to change it.

As you know the CI/CD should be automatic system 
Nobody want to change it every months or weeks

That why i want to make CI/CD script can automatically find the lastest SDK version on the build server.

And I have answer my question self here
https://github.com/fastlane/fastlane/discussions/17670#discussioncomment-148450

**Reply by rogerluan** [reply]
Sorry I didn't understand before. I thought you wanted to do it the other way around: change & revert changes only locally, not in the CI server. 

Glad you found the solution, though! 🎉
